### PR TITLE
seafile-client: fix build

### LINKED ISSUE #122505: seafile-client broken on unstable/master
**Describe the bug**
seafile-client is broken on unstable and master

**To Reproduce**
```
nix-build -A seafile-client
```

Build log: [build-log.txt](https://github.com/NixOS/nixpkgs/files/6455476/build-log.txt)

### PR DESCRIPTION
**Pull request by eduardosm**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/122505
ZHF: #122042

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122506` at fc39dac3 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>seafile-client</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/networking/seafile-client/default.nix:41:5:
```
   |
41 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/seafile-client/default.nix [hunk 1]
```diff
@@ -13,6 +13,15 @@ mkDerivation rec {
     sha256 = "2vV+6ZXjVg81JVLfWeD0UK+RdmpBxBU2Ozx790WFSyw=";
   };
 
+  patches = [
+    # Fix #122505
```
Comment (jonringer) [inline pkgs/applications/networking/seafile-client/default.nix:17]: do you mind expanding on the nature of the patch?
```suggestion
    # Fix issue with <summary>, fixes #122505
```

Comment (eduardosm) [reply pkgs/applications/networking/seafile-client/default.nix]: Done

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 122506 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>seafile-client</li>  </ul></details>

**Pull request merged**